src/Altinn.App.Api/Controllers/DataTagsController.cs

```diff
@@ -179,4 +192,110 @@ [FromRoute] string tag
 
         return NoContent();
     }
+
+    /// <summary>
+    /// Sets a set of tags on a data element.
+    /// </summary>
+    /// <param name="org">The short name for the application owner.</param>
+    /// <param name="app">The name of the application.</param>
+    /// <param name="instanceOwnerPartyId">The party id of the owner of the instance.</param>
+    /// <param name="instanceGuid">The id of the instance.</param>
+    /// <param name="dataGuid">The id of the data element.</param>
+    /// <param name="setTagsRequest">The request body.</param>
+    /// <param name="ignoredValidators">comma separated string of validators to ignore.</param>
+    /// <param name="language">The currently active user language.</param>
+    [HttpPut]
+    [Authorize(Policy = AuthzConstants.POLICY_INSTANCE_WRITE)]
+    [ProducesResponseType(typeof(SetTagsResponse), StatusCodes.Status200OK)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(typeof(ProblemDetails), StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<SetTagsResponse>> SetTags(
+        [FromRoute] string org,
+        [FromRoute] string app,
+        [FromRoute] int instanceOwnerPartyId,
+        [FromRoute] Guid instanceGuid,
+        [FromRoute] Guid dataGuid,
+        [FromBody] SetTagsRequest? setTagsRequest,
+        [FromQuery] string? ignoredValidators = null,
+        [FromQuery] string? language = null
+    )
+    {
+        if (!ModelState.IsValid || setTagsRequest is null)
+        {
+            return BadRequest(ModelState);
```

Comment:
What does ModelState do here? I think we should use `ProblemDetails` like it seems we've standardized on (that goes for the other responses below as well)

Reply:
Probably CodeQL suggestion because the controller is not marked with `[ApiController]` and thus the framework will call your action method even if not all parameters could be set to valid values from the request.

I think it would be wise to remove this block and add `[ApiController]` instead.